src/app/index.ts

```diff
@@ -156,41 +219,84 @@ module.exports = class TheiaPlugin extends Base {
             pluginSourcePath: pluginName + '-' + pluginType + '-plugin.ts',
             pluginDistPath: pluginName + '-' + pluginType + '-plugin.js',
             isFrontend: (options.pluginType === 'frontend'),
-            isBackend: (options.pluginType === 'backend')
+            isBackend: (options.pluginType === 'backend'),
+            template: options.template,
+            sample: options.sample,
         };
         options.params = this.params;
 
     }
 
     writing() {
+        this._writeBase();
```

Comment:
I would avoid to use _ for method names and we can let protected for these methods (if someone may want to extend class, etc)

Reply:
I also don't like '_' in method names, but yoman will run all methods in generated class, unless method name dont starts with '_', see http://yeoman.io/authoring/running-context.html for more details.

Reply:
should we use then another class/object to delegate to it ?

Reply:
Yes, I delegate to other class